docs/FeatureOverview.md

```diff
@@ -649,3 +649,66 @@ withCityInfo.select(
 ```scala mdoc:invisible
 spark.stop()
 ```
+
+## SparkExtension - Typed Literals push down
+
+If your queries could benefit from enhanced predicate push down when using Frameless typed literals you can register the FramelessSparkExtension:
+
+```scala mdoc
+import frameless.optimizer.FramelessExtension
+
+// With SparkConfig builders
+val sparkConfig = new org.apache.spark.SparkConf()
+  .setMaster("local[*]")
+  .setAppName("test")
+  .set("spark.ui.enabled", "false")
+sparkConfig.set("spark.sql.extensions", classOf[FramelessExtension].getName)
+
+val hostMode = "*"
+// With SparkSession.Builder()
+val sparkSession = org.apache.spark.sql.SparkSession.builder()
+  .config("spark.master", s"local[$hostMode]").config("spark.ui.enabled", false)
+  .config("spark.sql.extensions", classOf[FramelessExtension].getName)
+  .getOrCreate()
+```
+
+FramelessExtension injects the LiteralRule optimizer rule which unpacks FramelessLit into a Spark Literal without affecting encoding.  This can have large performance benefits when enabled via extensions.
```

Comment:
```suggestion
`FramelessExtension` injects the `LiteralRule` optimizer rule which unpacks `FramelessLit` into a Spark Literal without affecting encoding. This can have large performance benefits when enabled via extensions.
```